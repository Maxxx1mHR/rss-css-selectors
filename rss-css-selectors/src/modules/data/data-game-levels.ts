// export const dataExampleLayout = {
//   1: `<round data-id="1" class="shape round" data-tooltip="<round></round>"></round>
//       <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>`,

//   2: `<square data-id="1" class="shape square"  data-tooltip="<square></square>"></square>
//       <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
//       <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,

//   3: `<round data-id="1" id="border" class="shape round border" data-tooltip='<round id="border"></round>'></round>
//       <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
//       <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,

//   4: `<square data-id="1" class="shape square" data-tooltip="<square></square>"></square>
//       <round data-id="2" class="shape round" data-tooltip="<round></round>">
//         <point data-id="3" class="shape point" data-tooltip="<point></point>"></point>
//       </round>
//       <point data-id="4" class="shape point" data-tooltip="<point></point>"></point>`,

//   5: `<square data-id="1" class="shape square" data-tooltip="<square></square>">
//         <blue data-id="2" class="shape blue" data-tooltip="<blue></blue>"></blue>
//       </square>
//       <round data-id="3" id="border" class="shape round border" data-tooltip='<round id="border"></round>'>
//         <rectangle data-id="4" class="shape rectangle" data-tooltip="<rectangle></rectangle>"></rectangle>
//       </round>
//       <round data-id="5" class="shape round" data-tooltip="<round></round>">
//         <rectangle data-id="6" class="shape rectangle" data-tooltip="<rectangle></rectangle>"></rectangle >
//       </round>
//       `,
// };

// export const dataHTMLViewer = {
//   1: `
//   &ltdiv class="boat"&gt
//     <round data-id="1" class="code">&ltround/&gt</round>
//     <round data-id="2" class="code">&ltround/&gt</round>
//   &lt/div&gt
//   `,
//   2: `
//   &ltdiv class="boat"&gt
//     <square data-id="1" class="code">&ltsquare/&gt</square>
//     <round data-id="2" class="code">&ltround/&gt</round>
//     <square data-id="3" class="code">&ltsquare/&gt</square>
//   &lt/div&gt
//   `,
//   3: `
//   &ltdiv class="boat"&gt
//     <square data-id="1" id="border" class="code">&ltsquare id="border"/&gt</square>
//     <round data-id="2" class="code">&ltround/&gt</round>
//     <square data-id="3" class="code">&ltsquare/&gt</square>
//   &lt/div&gt
//   `,
//   4: `
//   &ltdiv class="boat"&gt
//     <square data-id="1" class="code">&ltsquare/&gt</square>
//     <round data-id="2" class="code">&ltround&gt
//       <point data-id="3" class="code">&ltpoint/&gt</point>
//     &lt/round&gt</round>
//     <point data-id="4" class="code">&ltpoint/&gt</point>
//   &lt/div&gt`,
//   5: `
//   &ltdiv class="boat"&gt
//     <square data-id="1" class="code">&ltsquare&gt
//       <blue data-id="2" class="code">&ltblue/&gt</blue>
//     &ltsquare/&gt</square><round data-id="3" id="border" class="code">
//     &ltround id="border"&gt
//       <rectangle data-id="4" class="code">&ltrectangle/&gt</rectangle>
//     &lt/round&gt</round>
//     <round data-id="5" class="code">&ltround&gt
//       <rectangle data-id="6" class="code">&lt/rectangle/&gt</rectangle >
//     &lt/round&gt</round>
//   &lt/div&gt`,
// };

export const dataGameLevels = [
  {
    level: 1,
    taskName: 'Type Selector',
    taskDescription: 'Select elements by their type',
    attribute: `<span>A</span>`,
    correction:
      'Selects all elements of type <span>A</span>. Type refers to the type of tag, so <span>&ltdiv&gt</span>, <span>&ltp&gt</span> and <span>&ltul&gt</span> are all different element types.',
    example: '<span>div</span> selects all <span>&ltdiv&gt</span> elements',
    dataExampleLayout: `<wheel data-id="1" class="shape wheel" data-tooltip="<wheel></wheel>"></wheel>
      <wheel data-id="2" class="shape wheel" data-tooltip="<wheel></wheel>"></wheel>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
    <wheel data-id="1" class="code">&wheel/&gt</wheel>
    <wheel data-id="2" class="code">&wheel/&gt</wheel>
  &lt/div&gt
      `,
  },
  {
    level: 2,
    taskName: 'Type Selector',
    taskDescription: 'Select elements by their type',
    attribute: `<span>A</span>`,
    correction:
      'Selects all elements of type <span>A</span>. Type refers to the type of tag, so <span>&ltdiv&gt</span>, <span>&ltp&gt</span> and <span>&ltul&gt</span> are all different element types.',
    example: '<span>div</span> selects all <span>&ltdiv&gt</span> elements',
    dataExampleLayout: `<map data-id="1" class="shape map"  data-tooltip="<map></map>"></map>
    <wheel data-id="2" class="shape wheel" data-tooltip="<wheel></wheel>"></wheel>
    <map data-id="3" class="shape map" data-tooltip="<map></map>"></map>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
    <map data-id="1" class="code">&map/&gt</map>
    <wheel data-id="2" class="code">&wheel/&gt</wheel>
    <map data-id="3" class="code">&map/&gt</map>
  &lt/div&gt
      `,
  },
  {
    level: 3,
    taskName: 'ID Selector',
    taskDescription: 'Select elements with an ID  ',
    attribute: `<span>#id</span>`,
    correction:
      'Selects the element with a specific <span>id</span>. You can also combine the ID selector with the type selector.',
    example: '<span>#cool</span> selects any element with <span>id="cool"</span> elements',
    dataExampleLayout: `<wheel data-id="1" id="first" class="shape wheel first" data-tooltip='<wheel id="first"></wheel>'></wheel>
      <wheel data-id="2" class="shape wheel" data-tooltip="<wheel></wheel>"></wheel>
      <map data-id="3" class="shape map" data-tooltip="<map></map>"></map>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
    <wheel data-id="1" id="first" class="code">&wheel id="first"/&gt</wheel>
    <wheel data-id="2" class="code">&wheel/&gt</wheel>
    <map data-id="3" class="code">&map/&gt</map>
  &lt/div&gt
      `,
  },
  {
    level: 4,
    taskName: 'Descendant Selector',
    taskDescription: 'Select an element inside another element',
    attribute: `<span>A  B</span>`,
    correction:
      'Selects all <span>B</span> inside of <span>A</span>. <span>B</span> is called a descendant because it is inside of another element.',
    example:
      '<span>p strong </span> selects all <span>&ltstrong&gt</span> elements that are inside of any <span>&lp&gt</span>',
    dataExampleLayout: `<map data-id="1" class="shape map" data-tooltip="<map></map>"></map>
    <wheel data-id="2" class="shape wheel" data-tooltip="<wheel></wheel>">
      <spyglass data-id="3" class="shape spyglass" data-tooltip="<spyglass></spyglass>"></spyglass>
    </wheel>
    <spyglass data-id="4" class="shape spyglass" data-tooltip="<spyglass></spyglass>"></spyglass>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
    <map data-id="1" class="code">&map/&gt</map>
    <wheel data-id="2" class="code">&wheel&gt
      <spyglass data-id="3" class="code">&spyglass/&gt</spyglass>
    &lt/wheel&gt</wheel>
    <spyglass data-id="4" class="code">&spyglass/&gt</spyglass>
  &lt/div&gt
      `,
  },
  {
    level: 5,
    taskName: 'No Name...',
    taskDescription: 'Combine the Descendant & ID Selectors',
    attribute: `<span>#id  A</span>`,
    correction: 'You can combine any selector with the descendent selector.',
    example:
      '<span>#cool span</span> selects all <span>&ltspan&gt</span> elements  that are inside of elements with <span>id="cool"</span>',
    dataExampleLayout: `<map data-id="1" class="shape map" data-tooltip="<map></map>">
    <sword data-id="2" class="shape sword" data-tooltip="<sword></sword>"></sword>
  </map>
  <wheel data-id="3" id="border" class="shape wheel border" data-tooltip='<wheel id="border"></wheel>'>
    <bottle data-id="4" class="shape bottle" data-tooltip="<bottle></bottle>"></bottle>
  </wheel>
  <wheel data-id="5" class="shape wheel" data-tooltip="<wheel></wheel>">
    <bottle data-id="6" class="shape bottle" data-tooltip="<bottle></bottle>"></bottle >
  </wheel>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
    <map data-id="1" class="code">&map&gt
      <sword data-id="2" class="code">&sword/&gt</sword>
    &map/&gt</map><wheel data-id="3" id="border" class="code">
    &wheel id="border"&gt
      <bottle data-id="4" class="code">&bottle/&gt</bottle>
    &lt/wheel&gt</wheel>
    <wheel data-id="5" class="code">&wheel&gt
      <bottle data-id="6" class="code">&lt/bottle/&gt</bottle >
    &lt/wheel&gt</wheel>
  &lt/div&gt
      `,
  },
];
