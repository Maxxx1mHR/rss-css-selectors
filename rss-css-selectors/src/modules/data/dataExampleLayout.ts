export const dataExampleLayout = {
  1: `<round data-id="1" class="shape round" data-tooltip="<round></round>"></round>
      <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>`,

  2: `<square data-id="1" class="shape square"  data-tooltip="<square></square>"></square>
      <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
      <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,

  3: `<round data-id="1" id="border" class="shape round border" data-tooltip='<round id="border"></round>'></round>
      <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
      <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,

  4: `<square data-id="1" class="shape square" data-tooltip="<square></square>"></square>
      <round data-id="2" class="shape round" data-tooltip="<round></round>">
        <point data-id="3" class="shape point" data-tooltip="<point></point>"></point>
      </round>
      <point data-id="4" class="shape point" data-tooltip="<point></point>"></point>`,

  5: `<square data-id="1" class="shape square" data-tooltip="<square></square>">
        <blue data-id="2" class="shape blue" data-tooltip="<blue></blue>"></blue>
      </square>
      <round data-id="3" id="border" class="shape round border" data-tooltip='<round id="border"></round>'>
        <rectangle data-id="4" class="shape rectangle" data-tooltip="<rectangle></rectangle>"></rectangle>
      </round>
      <round data-id="5" class="shape round" data-tooltip="<round></round>">
        <rectangle data-id="6" class="shape rectangle" data-tooltip="<rectangle></rectangle>"></rectangle >
      </round>
      `,
};

export const dataHTMLViewer = {
  1: `
  &ltdiv class="boat"&gt
    <round data-id="1" class="code">&ltround/&gt</round>
    <round data-id="2" class="code">&ltround/&gt</round>
  &lt/div&gt
  `,
  2: `
  &ltdiv class="boat"&gt
    <square data-id="1" class="code">&ltsquare/&gt</square>
    <round data-id="2" class="code">&ltround/&gt</round>
    <square data-id="3" class="code">&ltsquare/&gt</square>
  &lt/div&gt
  `,
  3: `
  &ltdiv class="boat"&gt
    <square data-id="1" id="border" class="code">&ltsquare id="border"/&gt</square>
    <round data-id="2" class="code">&ltround/&gt</round>
    <square data-id="3" class="code">&ltsquare/&gt</square>
  &lt/div&gt
  `,
  4: `
  &ltdiv class="boat"&gt
    <square data-id="1" class="code">&ltsquare/&gt</square>
    <round data-id="2" class="code">&ltround&gt
      <point data-id="3" class="code">&ltpoint/&gt</point>
    &lt/round&gt</round>
    <point data-id="4" class="code">&ltpoint/&gt</point>
  &lt/div&gt`,
  5: `
  &ltdiv class="boat"&gt
    <square data-id="1" class="code">&ltsquare&gt
      <blue data-id="2" class="code">&ltblue/&gt</blue>
    &ltsquare/&gt</square><round data-id="3" id="border" class="code">
    &ltround id="border"&gt
      <rectangle data-id="4" class="code">&ltrectangle/&gt</rectangle>
    &lt/round&gt</round>
    <round data-id="5" class="code">&ltround&gt
      <rectangle data-id="6" class="code">&lt/rectangle/&gt</rectangle >
    &lt/round&gt</round>
  &lt/div&gt`,
};

export const dataLevelsGame = [
  {
    level: 1,
    taskName: 'Type Selector',
    taskDescription: 'Select elements by their type',
    attribute: 'A',
    correction:
      'Selects all elements of type A. Type refers to the type of tag, so div, p and ul are all different element types.',
    example: 'div selects all div elements.',
    dataExampleLayout: `<round data-id="1" class="shape round" data-tooltip="<round></round>"></round>
      <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>`,
    dataHTMLViewer: `
      &ltdiv class="boat"&gt
        <round data-id="1" class="code">&ltround/&gt</round>
        <round data-id="2" class="code">&ltround/&gt</round>
      &lt/div&gt
      `,
  },
  {
    level: 2,
    taskName: 'Type Selector',
    taskDescription: 'Select elements by their type',
    attribute: 'A',
    correction:
      'Selects all elements of type A. Type refers to the type of tag, so div, p and ul are all different element types.',
    example: 'div selects all div elements.',
    dataExampleLayout: `<square data-id="1" class="shape square"  data-tooltip="<square></square>"></square>
    <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
    <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
      <square data-id="1" id="border" class="code">&ltsquare id="border"/&gt</square>
      <round data-id="2" class="code">&ltround/&gt</round>
      <square data-id="3" class="code">&ltsquare/&gt</square>
    &lt/div&gt
    `,
  },
  {
    level: 3,
    taskName: 'ID Selector',
    taskDescription: 'Select elements with an ID',
    attribute: '#id',
    correction: 'Selects the element with a specific id. You can also combine the ID selector with the type selector.',
    example: '#cool selects any element with id="cool"',
    dataExampleLayout: `<round data-id="1" id="border" class="shape round border" data-tooltip='<round id="border"></round>'></round>
    <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
    <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
      <square data-id="1" id="border" class="code">&ltsquare id="border"/&gt</square>
      <round data-id="2" class="code">&ltround/&gt</round>
      <square data-id="3" class="code">&ltsquare/&gt</square>
    &lt/div&gt
    `,
  },
];
