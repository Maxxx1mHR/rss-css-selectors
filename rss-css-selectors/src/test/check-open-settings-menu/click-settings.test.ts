import fs from 'fs';
import { fireEvent } from '@testing-library/dom';
import { dataGameLevels } from '../../modules/data/data-game-levels';
import MenuView from '../../modules/view/menu-view';

describe('render block menu', () => {
  it('check open menu click', () => {
    document.body.innerHTML = fs.readFileSync('src/index.html').toString();

    const level = 0;
    new MenuView(dataGameLevels, level);
    expect(document.body).toMatchSnapshot();

    expect(document.querySelector('.menu.active')).toBeFalsy();
    expect(document.querySelector('.menu')).toBeTruthy();
    expect(document.querySelector('.settings__open')).toBeTruthy();

    const setting = document.querySelector('.settings__open');
    if (setting) {
      fireEvent(setting, new MouseEvent('click', { bubbles: true, cancelable: true }));
    }
    expect(document.querySelector('.menu.active')).toBeTruthy();
  });
});
