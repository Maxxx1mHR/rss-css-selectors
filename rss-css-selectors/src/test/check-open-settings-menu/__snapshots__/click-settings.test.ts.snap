// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render block menu check open menu click 1`] = `
<body>
  

  
    
  <meta
    charset="UTF-8"
  />
  
    
  <meta
    content="IE=edge"
    http-equiv="X-UA-Compatible"
  />
  
    
  <meta
    content="width=device-width, initial-scale=1.0"
    name="viewport"
  />
  
    
  <title>
    RSS-CSS-Selectors
  </title>
  
    
  <link
    href="./assets/icons/favicon.ico"
    rel="shortcut icon"
    type="image/x-icon"
  />
  
    
  <link
    href="./assets/icons/favicon.ico"
    rel="icon"
    type="image/x-icon"
  />
  
  
  
    
  <div
    class="modal"
  >
    
      
    <div
      class="modal__wrapper"
    >
      
        
      <img
        alt="complete"
        class="modal__img"
        src="assets/img/complete.jpg"
      />
      
        
      <button
        class="modal__button"
      >
        
          
        <img
          alt="close modal"
          class="modal__skull-close"
          src="assets/icons/pirot/shapes/skull.png"
        />
        
        
      </button>
      
      
    </div>
    
    
  </div>
  
    
  <div
    class="settings"
  >
    <img
      class="settings__open"
    />
  </div>
  
    
  <div
    class="wrap"
  >
    
      
    <main
      class="main"
    >
      
        
      <div
        class="container"
      >
        
          
        <h1
          class="game__title"
        >
          RSS-CSS-Selectors
        </h1>
        
          
        <div
          class="game"
        >
          
            
          <div
            class="game__wrapper"
          />
          
            
          <div
            class="editor"
          >
            
              
            <div
              class="editor__css"
            >
              
                
              <div
                class="editor__header"
              >
                
                  
                <div
                  class="editor__header-name"
                >
                  CSS Editor
                </div>
                
                  
                <div
                  class="editor__header-extension"
                >
                  style.css
                </div>
                
                
              </div>
              
                
              <div
                class="editor__code-wrapper"
              >
                
                  
                <div
                  class="editor__numbers"
                >
                  
                    
                  <ul
                    class="editor__list"
                  >
                    
                      
                    <li
                      class="editor__item"
                    >
                      1
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      2
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      3
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      4
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      5
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      6
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      7
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      8
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      9
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      10
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      11
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      12
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      13
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      14
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      15
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      16
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      17
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      18
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      19
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      20
                    </li>
                    
                    
                  </ul>
                  
                  
                </div>
                
                  
                <div
                  class="editor__selector"
                />
                
                
              </div>
              
              
            </div>
            
              
            <div
              class="editor__html"
            >
              
                
              <div
                class="editor__header"
              >
                
                  
                <div
                  class="editor__header-name"
                >
                  HTML Viewer
                </div>
                
                  
                <div
                  class="editor__header-extension"
                >
                  table.html
                </div>
                
                
              </div>
              
                
              <div
                class="editor__code-wrapper"
              >
                
                  
                <div
                  class="editor__numbers"
                >
                  
                    
                  <ul
                    class="editor__list"
                  >
                    
                      
                    <li
                      class="editor__item"
                    >
                      1
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      2
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      3
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      4
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      5
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      6
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      7
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      8
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      9
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      10
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      11
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      12
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      13
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      14
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      15
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      16
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      17
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      18
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      19
                    </li>
                    
                      
                    <li
                      class="editor__item"
                    >
                      20
                    </li>
                    
                    
                  </ul>
                  
                  
                </div>
                
                  
                <pre
                  class="editor__code"
                />
                
                
              </div>
              
              
            </div>
            
            
          </div>
          
          
        </div>
        
        
      </div>
      
      
    </main>
    
      
    <footer
      class="footer"
    >
      
        
      <div
        class="container footer__wrapper"
      >
        
          
        <a
          href="https://github.com/Maxxx1mHR"
        >
          
            
          <img
            alt="GitHub"
            class="footer__img-git"
            src="assets/icons/interaction/github.png"
          />
          
          
        </a>
        
          
        <div
          class="footer__year"
        >
          2023
        </div>
        
          
        <a
          href="https://rs.school/js/"
        >
          
            
          <img
            alt="rs school"
            class="footer__img-rsschool"
            src="assets/icons/interaction/rs_school_js.svg"
          />
          
          
        </a>
        
        
      </div>
      
      
    </footer>
    
    
  </div>
  
    
  <div
    class="menu"
  >
    
      
    <div
      class="menu__block"
    >
      
        
      <div
        class="menu__header"
      >
        <div
          class="menu__level"
        >
          Level 1 of 10
        </div>
        <div
          class="menu__level-switch"
        >
          <img
            class="menu__prev"
          />
          <img
            class="menu__next"
          />
        </div>
      </div>
      
        
      <div
        class="menu__task"
      >
        
          
        <div
          class="menu__description"
        >
          <div
            class="menu__description-title"
          >
            Type Selector
          </div>
          <div
            class="menu__description-task"
          >
            Select elements by their type
          </div>
          <div
            class="menu__description-attribute"
          >
            <span>
              A
            </span>
          </div>
          <div
            class="menu__description-correction"
          >
            Selects all elements of type 
            <span>
              A
            </span>
            . Type refers to the type of tag, so 
            <span>
              &lt;div&gt;
            </span>
            , 
            <span>
              &lt;p&gt;
            </span>
             and 
            <span>
              &lt;ul&gt;
            </span>
             are all different element types.
          </div>
        </div>
        
          
        <div
          class="menu__example"
        >
          <div
            class="menu__example-title"
          >
            Example
          </div>
          <div
            class="menu__example-solution"
          >
            <span>
              div
            </span>
             selects all 
            <span>
              &lt;div&gt;
            </span>
             elements
          </div>
        </div>
        
        
      </div>
      
        
      <div
        class="menu__levels"
      >
        
          
        <div
          class="menu__wrapper"
        >
          <div
            class="menu__levels-title"
          >
            Chose the level
          </div>
          <img
            class="menu__reset"
          />
        </div>
        
          
        <ul
          class="menu__levels-list"
        >
          <li
            class="menu__levels-item menu__current-level"
            id="1"
          >
            Level 1 - Type Selector
          </li>
          <li
            class="menu__levels-item"
            id="2"
          >
            Level 2 - Type Selector
          </li>
          <li
            class="menu__levels-item"
            id="3"
          >
            Level 3 - ID Selector
          </li>
          <li
            class="menu__levels-item"
            id="4"
          >
            Level 4 - Descendant Selector
          </li>
          <li
            class="menu__levels-item"
            id="5"
          >
            Level 5 - Descendant ID Selector
          </li>
          <li
            class="menu__levels-item"
            id="6"
          >
            Level 6 - Comma Combinator and attribute selector
          </li>
          <li
            class="menu__levels-item"
            id="7"
          >
            Level 7 - First of Type Selector
          </li>
          <li
            class="menu__levels-item"
            id="8"
          >
            Level 8 - Last of Type Selector & Nth of Type Selector
          </li>
          <li
            class="menu__levels-item"
            id="9"
          >
            Level 9 - Nth сhild range
          </li>
          <li
            class="menu__levels-item"
            id="10"
          >
            Level 10 - Nth of Type Selector
          </li>
        </ul>
        
        
      </div>
      
        
      <div
        class="menu__close"
      >
        <img
          class="close"
        />
      </div>
      
      
    </div>
    
    
  </div>
  
  


</body>
`;
